keps/sig-node/127-user-namespaces/README.md

```diff
@@ -363,88 +549,463 @@ Note this section is a WIP yet.
+burden and latency.
+
+The KEP must have a approver from the
+[`prod-readiness-approvers`](http://git.k8s.io/enhancements/OWNERS_ALIASES)
+team. Please reach out on the
+[#prod-readiness](https://kubernetes.slack.com/archives/CPNHUMN74) channel if
+you need any help or guidance.
+-->
+
 ### Feature Enablement and Rollback
 
-* **How can this feature be enabled / disabled in a live cluster?**
-  - [ ] Feature gate
-    - Feature gate name:
-    - Components depending on the feature gate:
+<!--
+This section must be completed when targeting alpha to a release.
+-->
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+<!--
+Pick one of these and delete the rest.
+
+Documentation is available on [feature gate lifecycle] and expectations, as
+well as the [existing list] of feature gates.
 
-* **Does enabling the feature change any default behavior?**
+[feature gate lifecycle]: https://git.k8s.io/community/contributors/devel/sig-architecture/feature-gates.md
+[existing list]: https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/
+-->
+
+- [x] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: UserNamespacesSupport
+  - Components depending on the feature gate: kubelet, kube-apiserver
+
+###### Does enabling the feature change any default behavior?
+
+No.
+
+<!--
+Any change of default behavior may be surprising to users or break existing
+automations, so be extremely careful here.
+-->
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes. Pods that were created and opted-in to use user namespaces, will have to be
+recreated to run again without user namespaces.
+
+Other than that, it is safe to disable once it was enabled.
+
+<!--
+Describe the consequences on existing workloads (e.g., if this is a runtime
+feature, can it break the existing applications?).
 
-* **Can the feature be disabled once it has been enabled (i.e. can we roll back
-  the enablement)?**
+Feature gates are typically disabled by setting the flag to `false` and
+restarting the component. No other changes should be necessary to disable the
+feature.
 
-* **What happens if we reenable the feature if it was previously rolled back?**
+NOTE: Also set `disable-supported` to `true` or `false` in `kep.yaml`.
+-->
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+Pods will have to be re-created to use the feature.
+
+###### Are there any tests for feature enablement/disablement?
+
+We will add.
+
+We will test for when the field pod.spec.HostUsers is set to true, false
+and not set. All of this with and without the feature gate enabled.
 
-* **Are there any tests for feature enablement/disablement?**
+We will also unit test that, if pods were created with the new field
+pod.specHostUsers, then if the featuregate is disabled all works as expected (no
+user namespace is used).
```

Comment:
>> I would also like to see tests (integration?) that enable/disable the feature in the middle of the test and check what happens.

> [@wojtek-t](https://github.com/wojtek-t) do you have any suggestion on which feature gate to look for tests that enable/disable the feature in the middle of the test?
>
>Or just updating the KEP with your answer is enough for the KEP?

It's related to one of my comments above too - I would like to see that e.g. we can delete pods using the userns after the feature gate is disabled. So basically scenarios like:
- kubelet has feature gate on and we create a pod using userns
- we restart the kubelet and switch off feature gate
- we delete the pod and check if it was really deleted
[Something like that.]

There are probably other testcases we may have, so just for now adding a test that tests like that will get added is fine for Alpha.

Reply:
Perfect. I added it, as you suggested in the other comments, to the integration tests. As I already mentioned there, I'm not clear if we would be able to test all there or we will need an e2e test. But we will see that :)